### PR TITLE
LPS-195857 If the full name is empty then use the default username instead

### PR DESCRIPTION
**Pull request by MirnaGama**
### Steps to Reproduce:

1. Add a wiki widget to any page.
2. Configure a Node to operate wikis.
3. Create a wiki page in configured node
- Page Name: Program Test
- Version: 1.0
- Created by: Test Test

✅ Expected Behavior:
I would be able to have the users show up in the User Column of the Wiki Widget.

❌ Actual Behavior:
I am unable to have a user populate in the User Column of the Wiki Widget.

### Results of investigation:
It was discovered that the reason the user name was empty in the table row is because the get username method by the Portal util class returns an empty string if the user's full name is empty instead of returning the default name.

### CODE REVIEW

**Comment by MirnaGama** [comment]
ci:stop

**Comment by MirnaGama** [comment]
ci:test:sf

**Comment by MirnaGama** [comment]
ci:test:relevant

**Comment by MirnaGama** [comment]
ci:stop

**Comment by MirnaGama** [comment]
ci:test:sf

**Comment by MirnaGama** [comment]
ci:test:relevant

**Comment by MirnaGama** [comment]
ci:stop

**Comment by MirnaGama** [comment]
ci:test:sf

**Comment by MirnaGama** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@MirnaGama No regressions found. Testing passed following ticket steps.

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java [hunk 1]
```diff
@@ -5665,7 +5665,12 @@ public String getUserName(
 
 		if (user != null) {
 			if (userAttribute.equals(UserAttributes.USER_NAME_FULL)) {
-				userName = user.getFullName();
+				if ((user.getFullName() != null) ||
+					!user.getFullName(
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/util/PortalImpl.java:5669]: @MirnaGama Chaining is basically avoided in Liferay, and `==null` + `isEmpty` check can be simplified by using `Validator.isBlank()`.

So, please do something like

```
String userFullName = user.getFullName();

if (Validator.isNotBlank(userFullName)) {
    userName = userFullName;
}
```

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java [hunk 2]
```diff
@@ -5665,7 +5665,12 @@ public String getUserName(
...
+					).isEmpty()) {
+
+					userName = user.getFullName();
+				}
 			}
 			else {
 				userName = user.getScreenName();
```
Comment (dantewang) [inline portal-impl/src/com/liferay/portal/util/PortalImpl.java:5676]: @MirnaGama Does this branch suffer from the same issue?

**Review by dantewang — CHANGES_REQUESTED** [review]

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java (hunk 2 above)
Comment (MirnaGama) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java:5676]: It doesn't. 
When the method `getUserName(long userId, String defaultUserName) ` is called in `userName = getUserName(userId, userName)`, it explicitly sets the userAttribute value to _**USER_NAME_FULL**_.

**Comment by MirnaGama** [comment]
ci:test:sf

**Comment by MirnaGama** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java (hunk 1 above)
Comment (MirnaGama) [reply portal-impl/src/com/liferay/portal/util/PortalImpl.java]: Done ! 

https://github.com/liferay-core-infra/liferay-portal/pull/5534/commits/dc136323cc28ed9f86c17f5fb21e1ff3276a1f94

**Comment by catarineaguiar** [comment]
no regressions found. Manual test passed when following the ticket steps

**Review by dantewang — APPROVED** [review]

**Comment by shuyangzhou** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/5583

**Pull request closed without merging**